lib/promscrape/scrapework.go

```diff
@@ -1050,12 +1068,17 @@ func (wc *writeRequestCtx) addRow(cfg *ScrapeWork, r *parser.Row, timestamp int6
 	wc.labels = promrelabel.FinalizeLabels(wc.labels[:labelsLen], wc.labels[labelsLen:])
 	if len(wc.labels) == labelsLen {
 		// Skip row without labels.
-		return
+		return false
```

Comment:
This changes the behavior. If before series with no labels were just ignored, now the whole scrape will be rejected. And it will be rejected with false error "the response from %q exceeds label_limit..."

I'd suggest to change tryAddRow signature to return error. Then caller can differentiate between actual errors and nils